experimental/ir/lower_walk.go

```diff
@@ -150,6 +155,9 @@ func (w *walker) newType(def ast.DeclDef, parent any) Type {
 		)
 	}
 
+	file := &w.Context().file
+	file.types = append(file.types, w.Context().arenas.types.Compress(raw))
```

Comment:
This should only be done for top-level types, so it needs to go in an else block connected to the if immediately preceeding this.

I need to add separate tracking for non-top-level types... but I already have that in another PR, so you can leave it off for now.